packages/components/src/ThumbnailGallery/ThumbnailGallery.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from "react";
+import styles from "./ThumbnailGallery.css";
+import { FormatFile } from "../FormatFile";
+import { LightBox } from "../LightBox";
+
+interface Attachment {
+  /**
+   * The name of the file.
+   */
+  readonly name: string;
+
+  /**
+   * The [MIME](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types) type of the file
+   */
+  readonly type: string;
+
+  /**
+   * The size of the file in bytes.
+   */
+  readonly size: number;
+
+  /**
+   * The data url of the file.
+   */
+  url: string;
+}
+
+interface ThumbnailGalleryProps {
+  /**
+   * Takes list of details about the files to display as thumbnails.
+   */
+  readonly attachments: [Attachment];
+
+  /**
+   * Flag to show images in a LightBox on click
+   */
+  readonly openGalleryOnClick?: boolean;
+}
+
+export function ThumbnailGallery({ attachments }: ThumbnailGalleryProps) {
+  const [open, setOpen] = useState(false);
+  const [imageIndex, setImageIndex] = useState(0);
+
+  return (
+    <div className={styles.thumbnailContainer}>
+      {attachments &&
+        attachments.map((attachment, index) => {
+          return (
+            <div
+              key={`attachment-${index}`}
```

Comment:
I know there's warnings from React about using the `index` as the key value. I also know there's cases where it's acceptable to do. I can't say for sure which is applicable here, but something to keep in mind